src/test/java/org/sqlite/SerializeTest.java

```diff
@@ -0,0 +1,176 @@
+package org.sqlite;
+
+import org.junit.jupiter.api.Test;
+
+import java.nio.ByteBuffer;
+import java.sql.*;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+
+public class SerializeTest {
+
+    private ByteBuffer serialize() throws SQLException {
+        try (SQLiteConnection connection = (SQLiteConnection) DriverManager.getConnection("jdbc:sqlite:")) {
+            execute(connection, "ATTACH ? AS ?", ":memory:", "a_schema");
+            execute(connection, "CREATE TABLE a_schema.a_table (x integer)");
+            execute(connection, "INSERT INTO a_schema.a_table (x) values (?)", 1007);
+            return connection.serialize("a_schema");
+        }
+    }
+
+    @Test
+    public void testSerializeDeserialize() throws SQLException {
+        ByteBuffer bb = serialize();
+        deserializeAndAssert(bb);
+    }
+
+    private void deserializeAndAssert(ByteBuffer bb) throws SQLException {
+        try (SQLiteConnection connection = (SQLiteConnection) DriverManager.getConnection("jdbc:sqlite:")) {
+            execute(connection, "ATTACH ? AS ?", ":memory:", "another_schema");
+            connection.deserialize("another_schema", bb);
+            assertThat(fetch(connection, "SELECT * FROM another_schema.a_table")).isEqualTo(1007);
+        }
+    }
+
+    @Test
+    public void testSerializeByteArrayDeserialize() throws SQLException {
+        ByteBuffer bb = serialize();
+        assertThat(bb.isDirect()).isTrue();
+        byte[] buff = new byte[bb.remaining()];
+        bb.get(buff);
+        try {
+            deserializeAndAssert(ByteBuffer.wrap(buff));
+            fail("exception expected");
+        } catch (IllegalArgumentException ignore) {
+            // has to be a direct buffer
+        }
```

Comment:
probably better to use `assertThatThrownBy`

Reply:
done